www/content/workstation/upgrade_labs.md

```diff
@@ -0,0 +1,454 @@
+If you can't locate a cookbook, do not download it from an external source, such as the public Chef Supermarket. The cookbook version in your development environment must match the version on your node. [As a last resort](/upgrade_labs/#cookbooks-on-the-chef-server), the Upgrade Lab can get copies of your cookbooks from the Chef Infra Server during the `capture` phase.
+
+Likely cookbook locations:
+
+* Checked into your version control system
+* On a private Supermarket installation
+* In an existing [cookbook development directory](/upgrade_labs/#expected-cookbook-directory-layout)
+
+##### Cookbook Directory Layout
+
+If you have access to cookbook sources, it is simplest if you have the cookbooks stored in one parent directory, similar to this:
+
+```
+/Users/user_name/my_cookbooks/
+  ├── cron
+  │   ├── .git/   # Or other version control bookkeeping
+  │   ├── recipes/
+  │   ├──...
+  │   └── metadata.rb
+  └── chef-client
+  │   ├── .cvs/   # Or other version control bookkeeping
+  │   ├── recipes/
+  │   ├──...
+  │   └── metadata.rb
+  └── my_custom_cookbook
+      ├── .git/   # Or other version control bookkeeping
+      ├── recipes/
+      ├──...
+      └── metadata.rb
+```
+
+The Upgrade Lab works if you have cookbooks in different locations, but it involves more prompting from the command line.
+
+## Upgrade Lab
+
+### Inventory your system with Chef Reports
+
+We recommend starting the upgrade process on a node with a simpler a simple setup, such as one with fewer and simpler cookbooks.
+The `chef report` commands make a survey your nodes and cookbooks. Use the reports to identify a good starting place.
+
+#### chef report nodes
+
+Use `chef report nodes SERVER` command to create a report of the nodes in your system from Ohai data. The command:
+
+* Prints a report summary to the screen
+* Saves the report to the `.chef-workstation/reports/` directory.
+
+Create a node report from your development workstation by running:
+
+```shell
+chef report nodes -p old-server
+```
+
+Which outputs a node report:
+
+```output
+Analyzing nodes...
+
+-- REPORT SUMMARY --
+
+            Node Name             Chef Version    Operating System     Number Cookbooks
+--------------------------------+--------------+---------------------+-------------------
+  node-01                         12.22.5        windows v10.0.14393                 18
+  node-02                         12.22.5        windows v10.0.14393                 18
+  node-03                         12.22.5        windows v10.0.14393                  5
+  node-04                         12.18.31       windows v6.3.9600                    5
+Nodes report saved to /Users/user_name/.chef-workstation/reports/nodes-20200324135111.txt
+```
+
+#### chef report cookbooks
+
+Use `chef report cookbooks SERVER` command to create a report of the cookbooks in your system from Ohai data. The command:
```

Comment:
`chef report cookbooks -p PROFILE` - PROFILE is the name of a profile in the credentials file